### DISCUSSION TITLE
Bounding box in global coordinate

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#1485] [General] [answered]
I want to get element bounding box in global coordinate system.  It seems element BBoxLow, BBoxHigh are in local coordinates.  I can get origin, yaw, pitch and roll for the element - can someone point me to code that shows how to create a transformation matrix or a function to convert from local to global coordinates.

Thank you!

**Comment by pmconne** [comment, marked as answer]
The bounding box, origin, and orientation define the element's [Placement](https://www.itwinjs.org/reference/imodeljs-common/geometry/placement/).
Both `Placement3d` and `Placement2d` have a [calculateRange](https://github.com/imodeljs/imodeljs/blob/a18c75ddcf715f6c6feea080790b8783ca581b51/core/common/src/geometry/Placement.ts#L85) method that will give you the bounding box in world coordinates.
They can also supply a [Transform to world coordinates](https://github.com/imodeljs/imodeljs/blob/a18c75ddcf715f6c6feea080790b8783ca581b51/core/common/src/geometry/Placement.ts#L57).

**Reply by ovaisa** [reply]
Thank you Paul, it worked!  Much appreciate your support.

For benefit of others might need this in future, here is my code snippet

// Reference: frontend-sample-showcase>clash-review-sample>ClashReviewApp.tsx => calcClashCenter{}
 { 
    ...
    let range: any;
    const elemProps = (await iModel.elements.getProps([elementId])) as GeometricElement3dProps[];
    if (elemProps.length !== 0) {
        elemProps.forEach((prop: GeometricElement3dProps) => {
        const placement = Placement3d.fromJSON(prop.placement);
        range = placement.calculateRange();
      });
    }
    ...
 }
